nixos/modules/programs/steam.nix

```diff
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.steam;
+in {
+  options.programs.steam = {
+    enable = mkEnableOption "steam";
+    hardware = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable steam-hardware udev rules
+      '';
```

Comment:
I don't really get making another option for this that just enables `hardware.steam-hardware.enable`.
Can we remove this? You'll need to readd the bit about steam-hardware into the docs.

Reply:
In case it becomes more, for convenience, etc

Didn't want to move the hardware bits here as otherwise they'd be behind an .enable

Initially wanted to make hardware also enabled by default but there was some disagreement

Reply:
`programs.steam.hardware = true` -> `hardware.steam-hardware.enable = true;`
This added option doesn't do anything though. We could move the declaration of the steam-hardware option to this file, but that breaks how we have a whole directory in modules for hardware stuff.